exec/docker_swarm_stack_status_test.go

```diff
@@ -0,0 +1,148 @@
+		// 	ComposeFile:           "status_test_files/removing.yml",
+		// 	ExpectedStatus:        libstack.StatusRemoving,
+		// },
+		// {
+		// 	TestName:              "failed",
+		// 	ComposeFile:           "status_test_files/failed.yml",
+		// 	ExpectedStatus:        libstack.StatusError,
+		// 	ExpectedStatusMessage: true,
+		// },
+		// {
+		// 	TestName:              "removed",
+		// 	ComposeFile:           "status_test_files/removed.yml",
+		// 	ExpectedStatus:        libstack.StatusRemoved,
+		// },
+	}
+
+	w := setupSwarmStackDeployer(t)
+	ctx := context.Background()
+	dir := t.TempDir()
+
+	for _, testCase := range testCases {
+		t.Run(testCase.TestName, func(t *testing.T) {
+			projectName := testCase.TestName
+
+			composeFileName := fmt.Sprintf("docker-compose-%s.yml", projectName)
+			composeFilePath := filepath.Join(dir, composeFileName)
+			f, _ := os.Create(composeFilePath)
+			_, err := f.WriteString(testCase.FileContent)
+			if err != nil {
+				t.Fatalf("[test: %s] Failed to write compose file: %v", testCase.TestName, err)
+			}
+
+			err = w.Deploy(ctx, projectName, []string{composeFilePath}, agent.DeployOptions{})
+			if err != nil {
+				t.Fatalf("[test: %s] [path: %s] Failed to deploy compose file: %v", testCase.TestName, composeFilePath, err)
+			}
+
+			status, statusMessage, err := waitForStatus(w, ctx, projectName, libstack.StatusRunning)
+			if err != nil {
+				t.Fatalf("[test: %s] Failed to get compose project status: %v", testCase.TestName, err)
+			}
+
+			if status != testCase.ExpectedStatus {
+				t.Fatalf("[test: %s] Expected status: %s, got: %s", testCase.TestName, testCase.ExpectedStatus, status)
+			}
+
+			if testCase.ExpectedStatusMessage && statusMessage == "" {
+				t.Fatalf("[test: %s] Expected status message but got empty", testCase.TestName)
+			}
+
+			err = w.Remove(ctx, projectName, nil, agent.RemoveOptions{})
+			if err != nil {
+				t.Fatalf("[test: %s] Failed to remove compose project: %v", testCase.TestName, err)
+			}
+
+			status, statusMessage, err = waitForStatus(w, ctx, projectName, libstack.StatusRemoved)
+			if err != nil {
+				t.Fatalf("[test: %s] Failed to get compose project status: %v", testCase.TestName, err)
+			}
+
+			if status != libstack.StatusRemoved {
+				t.Fatalf("[test: %s] Expected stack to be removed, got %s", testCase.TestName, status)
+			}
+
+			if statusMessage != "" {
+				t.Fatalf("[test: %s] Expected empty status message: %s, got: %s", "", testCase.TestName, statusMessage)
+			}
+		})
+	}
+}
+
+func waitForStatus(deployer *DockerSwarmStackService, ctx context.Context, stackName string, requiredStatus libstack.Status) (libstack.Status, string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	statusCh := deployer.WaitForStatus(ctx, stackName, requiredStatus)
+	select {
+	case result := <-statusCh:
+		if result == "" {
+			return requiredStatus, "", nil
+		}
+
+		return libstack.StatusError, result, nil
+	case <-ctx.Done():
```

Comment:
This channel read contends with the the same one in `deployer.WaitForStatus()`, only one can succeed, I suggest removing it from here.

Reply:
:+1: